components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenuBuilder.kt

```diff
@@ -37,34 +38,53 @@ class WebExtensionBrowserMenuBuilder(
     private val store: BrowserStore,
     private val style: Style = Style(),
     private val onAddonsManagerTapped: () -> Unit = {},
-    private val appendExtensionSubMenuAtStart: Boolean = false
+    private val appendExtensionSubMenuAtStart: Boolean = false,
+    private val showAddonsInMenu: Boolean = true
 ) : BrowserMenuBuilder(items, extras, endOfMenuAlwaysVisible) {
 
+    private val finalList = items.toMutableList()
```

Comment:
I'm bit concern about making the `finalList`global in `WebExtensionBrowserMenuBuilder`, as multiple calls to `build` will persist previous values, for example, on line 78 [finalList[index] = menuItem](https://github.com/mozilla-mobile/android-components/pull/10564/files#diff-dccc8321d3ba7a093a75317f56364660611080c9f3e4e155688ce1bdc7609c96R78), we are altering `finalList` and keeping the altered list between calls to `build`, before we used to creating a new list between calls to `build`, now are keeping previous values between subsequent calls to `build`.

Reply:
If the intention is just to share the list between functions, we could pass the list as parameter, where we want to use it.